faster_whisper/transcribe.py

```diff
@@ -661,8 +672,14 @@ def generate_with_fallback(
 
             if not needs_fallback:
                 break
+        else:
+            # all failed, select the result with the highest log prob
+            if below_cr_threshold_results:
+                decode_result = max(below_cr_threshold_results, key=lambda x: x[3])
+            else:
+                decode_result = max(all_results, key=lambda x: x[3])
```

Comment:
The index should be 1.

You can write this in a single line (not sure if it is too long for the current style check):

```suggestion
            decode_result = max(below_cr_threshold_results or all_results, key=lambda x: x[1])
```